felix/fv/bpf_test.go

```diff
@@ -3842,3 +3846,46 @@ func checkServiceRoute(felix *infrastructure.Felix, ip string) bool {
 
 	return false
 }
+
+func bpfCheckIfPolicyProgrammed(felix *infrastructure.Felix, iface, hook, polName, action string) bool {
+	startStr := fmt.Sprintf("Start of policy %s", polName)
+	endStr := fmt.Sprintf("End of policy %s", polName)
+	actionStr := fmt.Sprintf("Start of rule action:\"%s\"", action)
+	var policyDbg bpf.PolicyDebugInfo
+	out, err := felix.ExecOutput("cat", bpf.PolicyDebugJSONFileName(iface, hook))
+	if err != nil {
+		return false
+	}
+	dec := json.NewDecoder(strings.NewReader(string(out)))
+	err = dec.Decode(&policyDbg)
+	if err != nil {
+		return false
+	}
+
+	hookStr := "tc ingress"
+	if hook == "ingress" {
+		hookStr = "tc egress"
+	}
```

Comment:
This should only apply to workload ifaces,right? Similarly as then calico-bpf dumps the policies. I think it is wrong in both cases.